src/SFA.DAS.AANHub.Api/Common/ActionResponseControllerBase.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.AANHub.Application.Mediatr.Common;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+
+namespace SFA.DAS.AANHub.Api.Common
+{
+    public class ActionResponseControllerBase : ControllerBase
+    {
+        private readonly ILogger<ActionResponseControllerBase> _logger;
+
+        public ActionResponseControllerBase(ILogger<ActionResponseControllerBase> logger) => _logger = logger;
+
+        protected IActionResult GetResponse<T>(ValidatedResponse<T> response, BaseRequestDetails requestDetails) where T : class
+        {
+            if (response.Result == null && response.Errors.Count == 0)
+            {
+                _logger.LogTrace("No data found for {controllerName}: {actionName}. Id: {id}",
```

Comment:
Forcing each GET method to populate the request details seems excessive for logging purpose only and I dont think this log adds any value. Application Insight will capture all these anyway. Also there is a less intrusive way to get this information if needed by using HttpRequest object.

Reply:
GET logs removed